### ISSUE TITLE
Add mailing list for Edward

### ISSUE THREAD

**Issue by dustinvtran** [blei-lab/edward#530]
This should be set up immediately. The [Gitter channel](http://gitter.im/blei-lab/edward) is growing with user questions that would benefit from a better platform.

The key decision is whether to use Discourse, StackOverflow, or Google Groups.

References
+ [OpenAI uses Discourse](https://discuss.openai.com).
+ TensorFlow uses [StackOverflow for support](https://stackoverflow.com/questions/tagged/tensorflow) and [mailing lists for discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/discuss).
+ Torch uses [Discourse](http://discuss.pytorch.org/) and announcement-based newletters.
+ Theano uses both mailing lists and StackOverflow.
+ Keras uses both mailing lists and StackOverflow.
+ [Dev vs user lists discussion in Stan](http://discourse.mc-stan.org/t/dev-vs-user-lists/).
+ Julia uses Discourse and Stack Overflow. [[link]](http://julialang.org/community/)
+ Rust uses Discourse and Stack Overflow. [[link]](https://www.rust-lang.org/en-US/community.html)

**Comment by dustinvtran** [comment]
Mailing lists (e.g, Google groups) are aged—the entire paradigm of following 50+ e-mails a day seems broken. So I think that rules them out.

Q&A platforms (e.g., StackOverFlow) are perfect for answering questions with clear answers and has excellent search features. However, the purposeful stifling of discussion makes it difficult to carry prolonged chats, answer questions with more opinion than fact, or build a community around.

I think we should use Discourse. With the caveat that it will never be a drop-in mailing list replacement, and Q&A in it may not scale if users end up referring back to frequently asked questions. If the time arises, we can look into adding these options later.

**Comment by JohnReid** [comment]
I agree mailing lists are broken. Stan has done so many things right but choosing google groups made it so much more time consuming to find answers to common problems. I much prefer Stack Overflow. I vote for keeping Gitter for discussion but using Stack Overflow for specific issues. Or perhaps github issues are good places to have focussed discussions. Having said all that I don't know anything about Discourse.

**Comment by ensemblearner** [comment]
IMHO, I don't think StackOverflow is a right medium to address general user concerns. 
Sure, if one has encountered a bug or anything in edward, then SO works fine but plenty of times, I have seen plenty of questions in SO downvoted and closed just because user is new and might not know/understand the paradigm just yet. 
Though if we maintain both gitter and SO (as suggested by @JohnReid) , then this issue is addressed.
But if we want to maintain just one platform, then definitely gitter over SO... 
Havent used Discourse.. Does google indexes conversation in Discourse?

**Comment by junpenglao** [comment]
I would suggest Discourse as well

**Comment by akucukelbir** [comment]
People seem to be happy with Discourse.

**Comment by dustinvtran** [comment]
> Does google indexes conversation in Discourse?

Fun story: I googled "discourse google index" and went through Meta Discourse's discussion to find this out. Then 5 minutes later I realized that I had already solved my problem, by successfully making the Google search in the first place.

> if we maintain both gitter and SO (as suggested by @JohnReid) , then this issue is addressed.

This seems right. Rust and Julia handle it this way, where user questions can exist on either Discourse or SO. Certainly user questions will percolate onto SO regardless, so maybe it's useful to officially support it.
